olm/catalogd/v1alpha1/clustercatalog_types.go

```diff
@@ -0,0 +1,223 @@
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="Image"
+	// +kubebuilder:validation:Required
+	Type SourceType `json:"type"`
+	// image is used to configure how catalog contents are sourced from an OCI image. This field must be set when type is set to "Image" and must be the only field defined for this type.
+	// +optional
+	Image *ImageSource `json:"image,omitempty"`
+}
+
+// ResolvedCatalogSource is a discriminated union of resolution information for a Catalog.
+// ResolvedCatalogSource contains the information about a sourced Catalog
+// +union
+// +kubebuilder:validation:XValidation:rule="self.type == 'Image' && has(self.image)",message="source type 'Image' requires image field"
+type ResolvedCatalogSource struct {
+	// type is a reference to the type of source the catalog is sourced from.
+	//
+	// It will be set to one of the following values: ["Image"].
+	//
+	// When this field is set to "Image", information about the resolved image source will be set in the 'image' field.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="Image"
+	// +kubebuilder:validation:Required
+	Type SourceType `json:"type"`
+	// image is a field containing resolution information for a catalog sourced from an image.
+	Image *ResolvedImageSource `json:"image"`
+}
+
+// ResolvedImageSource provides information about the resolved source of a Catalog sourced from an image.
+type ResolvedImageSource struct {
+	// ref contains the resolved sha256 image ref containing Catalog contents.
+	Ref string `json:"ref"`
+	// lastSuccessfulPollAttempt is the time when the resolved source was last successfully polled for new content.
+	LastSuccessfulPollAttempt metav1.Time `json:"lastSuccessfulPollAttempt"`
+}
+
+// ImageSource enables users to define the information required for sourcing a Catalog from an OCI image
+type ImageSource struct {
+	// ref is a required field that allows the user to define the reference to a container image containing Catalog contents.
+	// Examples:
+	//   ref: quay.io/operatorhubio/catalog:latest # image reference
+	//   ref: quay.io/operatorhubio/catalog@sha256:c7392b4be033da629f9d665fec30f6901de51ce3adebeff0af579f311ee5cf1b # image reference with sha256 digest
+	Ref string `json:"ref"`
+	// pollInterval is an optional field that allows the user to set the interval at which the image source should be polled for new content.
+	// It must be specified as a duration.
+	// It must not be specified for a catalog image referenced by a sha256 digest.
+	// Examples:
+	//   pollInterval: 1h # poll the image source every hour
+	//   pollInterval: 30m # poll the image source every 30 minutes
+	//   pollInterval: 1h30m # poll the image source every 1 hour and 30 minutes
+	//
+	// When omitted, the image will not be polled for new content.
+	// +kubebuilder:validation:Format:=duration
+	// +optional
+	PollInterval *metav1.Duration `json:"pollInterval,omitempty"`
```

Comment:
What is the resync period for the informers in the controller that will be processing this? Does a full resync of the informer cause this to poll?

I'm wondering if the resync of the informer (which might be 10m, 10h, 24h...) actually sets an upper bound for the poll interval